cpp/ql/src/jsf/4.21 Operators/AV Rule 165.ql

```diff
@@ -4,13 +4,22 @@
  * @kind problem
  * @id cpp/jsf/av-rule-165
  * @problem.severity warning
- * @tags reliability
+ * @precision low
+ * @tags correctness
+ *       reliability
+ *       external/jsf
  */
 import cpp
 
 // see MISRA Rule 5-3-2
 
 from UnaryMinusExpr ume
-where ume.getOperand().getUnderlyingType().(IntegralType).isUnsigned()
+where ume.getOperand().getFullyConverted().getUnderlyingType().(IntegralType).isUnsigned()
       and not ume.getOperand() instanceof Literal
-select ume, "The unary minus operator should not be applied to an unsigned expression."
+//select ume, "The unary minus operator should not be applied to an unsigned expression."
+select
+	ume,
+	ume.getOperand().getUnderlyingType(),
+	ume.getOperand().getFullyConverted().getUnderlyingType(),
+	ume.getUnderlyingType(),
+	ume.getFullyConverted().getUnderlyingType()
```

Comment:
It appears you left some test code behind here. Fortunately it was caught by PR-Checks.